packages/design-picker/src/components/style-variation-badges/index.tsx

```diff
@@ -16,32 +16,47 @@ interface BadgesProps {
 }
 
 const Badges: React.FC< BadgesProps > = ( {
+	className,
 	maxVariationsToShow = 4,
 	variations = [],
 	onMoreClick,
 	onClick,
 	selectedVariation,
 } ) => {
-	const isSelectedVariationDefault = isDefaultGlobalStylesVariationSlug( selectedVariation?.slug );
+	const isSelectedVariationDefault = isDefaultVariation( selectedVariation );
+	const { defaultVariation, styleVariations, colorVariations } = useMemo(
+		() => getGroupedVariations( variations ),
+		[ variations ]
+	);
+	// Use the color variations if the style variations are empty because we don't display color variations as palette section.
+	const currentStyleVariations = styleVariations.length > 0 ? styleVariations : colorVariations;
 	const variationsToShow = useMemo( () => {
-		return variations.slice( 0, maxVariationsToShow );
-	}, [ variations, maxVariationsToShow ] );
+		return currentStyleVariations.slice( 0, maxVariationsToShow - 1 );
```

Comment:
I updated the `maxVariationsToShow` to avoid confusion. The reason is this PR moved the default one of out the array so the badge now is showing `default s1 s2 s3 +x`

Reply:
> the badge now is showing `default s1 s2 s3 +x`

I still don't understand... TT5 is still not showing the default variation circle 🤔 

<img width="426" alt="image" src="https://github.com/user-attachments/assets/3807effb-974d-4983-ade4-fadc0ae73fb7" />

Reply:
it's another bug that the variation may have color `color-mix(in srgb, currentColor 20%, transparent)`...

Reply:
It should be fixed by [c15a9a8](https://github.com/Automattic/wp-calypso/pull/97808/commits/c15a9a842a32cd63780cb90799c47af476e60408)